lib/rom/array_dataset.rb

```diff
@@ -0,0 +1,12 @@
+require 'rom/enumerable_dataset'
+
+module ROM
+  class ArrayDataset < EnumerableDataset
+    forward(
+      :*, :+, :-, :compact, :compact!, :delete_if, :flatten, :flatten!, :keep_if,
+      :map!, :length, :pop, :reject, :reject!, :reverse, :reverse!, :sample,
+      :select!, :size, :shift, :shuffle, :shuffle!, :slice, :slice!, :sort!,
+      :sort_by!, :uniq, :uniq!, :unshift, :values_at
```

Comment:
@solnic – any feelings on the maintainability of this? (I am *not* suggesting we change it for this PR :grin: )